sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageUtils.java

```diff
@@ -369,4 +380,167 @@ public static <T> T describedToOrigin(DescribedType describedType) {
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setMessageId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setTo((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setReplyTo((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setLabel((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setSessionId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setReplyToSessionId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setContentType((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                Object properties = describedCorrelationFilter.next();
+                if (properties instanceof Map) {
+                    correlationFilter.getProperties().putAll((Map<String, ?>) properties);
+                }
+            }
+
+            return correlationFilter;
+        } else if (describedFilter.getDescriptor().equals(ServiceBusConstants.TRUE_FILTER_NAME)) {
+            return new TrueRuleFilter();
+        } else if (describedFilter.getDescriptor().equals(ServiceBusConstants.FALSE_FILTER_NAME)) {
+            return new FalseRuleFilter();
+        } else {
+            throw new UnsupportedOperationException("This client cannot support filter with descriptor: " + describedFilter.getDescriptor());
+        }
+
+        return null;
+    }
+
+    /**
+     * Get {@link RuleAction} from a {@link DescribedType}.
+     *
+     * @param describedAction A {@link DescribedType} with rule action information.
+     * @return A {@link RuleAction}.
+     */
+    private static RuleAction decodeRuleAction(DescribedType describedAction) {
+        if (describedAction.getDescriptor().equals(ServiceBusConstants.EMPTY_RULE_ACTION_NAME)) {
+            return null;
+        } else if (describedAction.getDescriptor().equals(ServiceBusConstants.SQL_RULE_ACTION_NAME)
+            && describedAction.getDescribed() instanceof Iterable) {
+            @SuppressWarnings("unchecked") Iterator<Object> describedSqlAction = ((Iterable<Object>) describedAction.getDescribed()).iterator();
+            if (describedSqlAction.hasNext()) {
+                return new SqlRuleAction((String) describedSqlAction.next());
+            }
+        }
+
+        return null;
```

Comment:
Same as comment above, just making sure we don't accidentally run into a NPE

Reply:
Checked, and we allow a null rule action.